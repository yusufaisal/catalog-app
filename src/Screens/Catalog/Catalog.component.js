import * as React from 'react';
import { View } from 'react-native';

import Header from '../../Components/Header/Header.component';
import CategoryList from '../../Components/CategoryList/CategoryList.component';
import ProductList from '../../Components/ProductList/ProductList.component';

import styles from './Catalog.component.styles';

import {
  getCategory,
  getProduct
} from '../../Utils/Network.utils';

const _renderHeader = () => (
  <Header title={'Shoes'}/>
);

const _renderCategoryList = (state) => (
  <CategoryList
    items={state.categories}
  />
);

const _renderProductList = (state) => (
  <ProductList
    products={state.products}
  />
)

const _getCategory = async (setState) => {
  try {
    const categories = await getCategory();
    const products = await getProduct();

    setState(prevState => ({
      ...prevState,
      products, 
      categories: categories
    }))
  } catch {
    // do nothing
  }
}
const useNetworkCall = (setState) => {
  React.useEffect(() => {
    _getCategory(setState);
  }, []);
};

const Catalog = () => {
  const [state, setState] = React.useState({
    products: [],
    categories: []
  })

  useNetworkCall(setState);

  return (
    <View style={styles.container}>
      {_renderHeader()}
      {_renderCategoryList(state)}
      {_renderProductList(state)}
    </View>
  )
}

export default Catalog