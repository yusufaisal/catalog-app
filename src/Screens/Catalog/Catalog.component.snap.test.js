import React from 'react';
import renderer from 'react-test-renderer';

jest
  .mock('../../Components/Header/Header.component', () => 'Header')
  .mock('../../Components/CategoryList/CategoryList.component', () => 'CategoryList')
  .mock('../../Components/ProductList/ProductList.component', () => 'ProductList')
  .mock('../../Utils/Network.utils', () => ({
    getCategory: jest.fn(),
    getProduct: jest.fn()
  }));

import Catalog from './Catalog.component';
import {
  getCategory,
  getProduct
} from '../../Utils/Network.utils';

describe('Catalog screen', () =>  {

  const categoryResp = [];
  const productResp = [];

  beforeEach(() => {
    jest.useFakeTimers();
    getCategory.mockReturnValueOnce(categoryResp);
    getProduct.mockReturnValueOnce(productResp);
  });

  it('should able to render screen properly', async () => {
    jest.runAllTimers();
    const tree = await renderer
      .create(<Catalog/>)
      .toJSON();
    jest.runAllTimers();

    expect(tree).toMatchSnapshot();
  })
})
