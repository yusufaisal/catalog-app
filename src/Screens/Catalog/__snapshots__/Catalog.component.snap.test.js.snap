// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog screen should able to render screen properly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "flex": 1,
    }
  }
>
  <Header
    title="Shoes"
  />
  <CategoryList
    items={Array []}
  />
  <ProductList
    products={Array []}
  />
</View>
`;
